Sources/ETCD/ETCDClient.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import GRPC
+import NIO
+import SwiftProtobuf
+
+
+public final class EtcdClient: Sendable {
+    private let host: String
+    private let port: Int
+    private var group: EventLoopGroup
+    private var connection: ClientConnection
+    private var client: Etcdserverpb_KVClient!
+
+    public init(host: String, port: Int, eventLoopGroup: EventLoopGroup) {
+        self.host = host
+        self.port = port
+        self.group = eventLoopGroup
+        self.connection = ClientConnection.insecure(group: self.group)
+            .connect(host: host, port: port)
+        self.client = Etcdserverpb_KVClient(channel: self.connection)
+    }
+
+    public func set(_ key: String, value: String) async throws {
```

Comment:
This function needs to become generic over the `ETCDValue` and we need a `ETCDKey` protocol as well which it can become generic over. So in the end the function signature must look like this `public func set(_ key: some ETCDKey, value: some ETCDValue) async throws`

Reply:
Quick question! Do I need a ETCDKey type. Memcache used a string as the key, should I keep it as a string? Should it be a similar protocol to ETCDValue, e.g. conforming over the same types?